### PR TITLE
feat(): Implementing gorgias connector

### PR DESCRIPTION
**Pull request by mohan-pogala**
# Purpose
This PR is to implement the Gorgias connector for the fidesops
# Changes
The PR contains the YAML changes for both config and dataset
-
# Checklist

- [ ] Applicable documentation updated (guides, quickstart, postman collections, tutorial, fidesdemo, [database diagram](https://github.com/ethyca/fidesops/blob/main/docs/fidesops/docs/development/update_erd_diagram.md))
- [ ] Good unit test/integration test coverage
- [ ] This PR contains a DB migration. If checked, the reviewer should confirm with the author that the [down_revision correctly references the previous migration](https://ethyca.github.io/fidesops/development/contributing_details/#alembic-migrations) before merging
- [ ] The `Run Unsafe PR Checks` label has been applied, and checks have passed, if this PR touches any external services

# Ticket

Fixes #

### CODE REVIEW

File: data/saas/config/gorgias_config.yml [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+  name: Gorgias SaaS Config
+  description: A sample schema representing the Gorgias connector for Fidesops
+  version: 0.0.1
+
+  connector_params:
+    - name: domain
+    - name: api_email
+    - name: api_key
+    - name: per_page
+    - name: email
```
Comment (galvana) [inline data/saas/config/gorgias_config.yml:12]: The `email` connector param can be removed here. 

In the legacy spec, we had `paramSpecs` of type `DATA_STORE` and `IDENTITY`. For Fidesops you can think of the `connector_params` as being the place to store the legacy equivalent of `DATA_STORE` paramSpecs (information provided by the user during configuration). `IDENTITY` params are handled separately in Fidesops as inputs to [privacy requests](https://ethyca.github.io/fidesops/guides/privacy_requests/#how-do-i-submit-a-privacy-request).

File: data/saas/dataset/gorgias_dataset.yml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  - fides_key: gorgias_connector_example
+    name: Gorgias Dataset
+    description: A sample dataset representing the Gorgias connector for Fidesops
+    collections:
+      - name: customer
+        fields:
+          - name: id
+            data_categories: [user.derived.identifiable.unique_id]
+            fidesops_meta:
+              primary_key: True
```
Comment (galvana) [inline data/saas/dataset/gorgias_dataset.yml:11]: Nice flagging of the `primary_key` for these collections.

File: data/saas/dataset/gorgias_dataset.yml [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+            fidesops_meta:
+              data_type: string
+      - name: tickets
+        fields:
+          - name: id
+            data_categories: [user.derived.identifiable.unique_id]
+            fidesops_meta:
+              primary_key: True
+            # TODO - Custom data_type wasn't supported at the time it was written
+            # TODO - Has to change the datatype to BIGINT
```
Comment (galvana) [inline data/saas/dataset/gorgias_dataset.yml:35]: In Python3 we can use the `int` type to store what we would consider a `BigInteger` in Java. But good callout!

**Review by galvana — COMMENTED** [review]
Overall good first pass based on the partial feature implementation we have for SaaS connectors in Fidesops. I appreciate you calling out the parts that didn't have a 1 to 1 mapping between the legacy spec and the Fidesops spec. I'll consider this a draft for now but we can merge it in once we fill in the missing pieces (like pagination and connector_param references in the request_params).

File: data/saas/dataset/gorgias_dataset.yml (hunk 2 above)
Comment (mohan-pogala) [reply data/saas/dataset/gorgias_dataset.yml:11]: Thank you

**Pull request closed without merging**